### DISCUSSION TITLE
Add CSS shadow parts support

### DISCUSSION THREAD

**Discussion by adi-heffer-folloze** [add2cal/add-to-calendar-button#541] [Ideas / Requests]
Hi all,

I'd like to suggest [CSS shadow part](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_shadow_parts) as a solution for external modifications of shadow dom styles.

For example: By adding the native attribute part="button" to the button elements, we should be able to select them in external CSS files with the simple pseudo selector `add-to-calendar-button::part(button)`.

I think that button, icon, label and list are parts worth exposing. It will make style customization much much easier and we could avoid long inline css that way.
What do you think?

**Comment by jekuer** [comment]
**I really like this!**
It also seems that this is now widely supported.
Would put it on the  nearer roadmap to extend options to alter the style from outside the shadow DOM (as suggested).
Will also give it even more attention with the next bigger refactoring as I see big potential for optimization here.

**Comment by jekuer** [comment]
Implemented with v2.6
